nav2_amcl/src/amcl_node.cpp

```diff
@@ -1113,6 +1124,203 @@ AmclNode::initParameters()
+      } else if (param_name == "update_min_d") {
+        d_thresh_ = parameter.as_double();
+      } else if (param_name == "z_hit") {
+        z_hit_ = parameter.as_double();
+        reinit_laser = true;
+      } else if (param_name == "z_max") {
+        z_max_ = parameter.as_double();
+        reinit_laser = true;
+      } else if (param_name == "z_rand") {
+        z_rand_ = parameter.as_double();
+        reinit_laser = true;
+      } else if (param_name == "z_short") {
+        z_short_ = parameter.as_double();
+        reinit_laser = true;
+      }
+    } else if (param_type == ParameterType::PARAMETER_STRING) {
+      if (param_name == "base_frame_id") {
+        base_frame_id_ = parameter.as_string();
+      } else if (param_name == "global_frame_id") {
+        global_frame_id_ = parameter.as_string();
+      } else if (param_name == "map_topic") {
+        map_topic_ = parameter.as_string();
+        reinit_map = true;
+      } else if (param_name == "laser_model_type") {
+        sensor_model_type_ = parameter.as_string();
+        reinit_laser = true;
+      } else if (param_name == "odom_frame_id") {
+        odom_frame_id_ = parameter.as_string();
+        reinit_laser = true;
+      } else if (param_name == "scan_topic") {
+        scan_topic_ = parameter.as_string();
+        reinit_laser = true;
+      } else if (param_name == "robot_model_type") {
+        robot_model_type_ = parameter.as_string();
+        reinit_odom = true;
+      }
+    } else if (param_type == ParameterType::PARAMETER_BOOL) {
+      if (param_name == "do_beamskip") {
+        do_beamskip_ = parameter.as_bool();
+        reinit_laser = true;
+      } else if (param_name == "tf_broadcast") {
+        tf_broadcast_ = parameter.as_bool();
+      } else if (param_name == "set_initial_pose") {
+        set_initial_pose_ = parameter.as_bool();
+      } else if (param_name == "always_reset_initial_pose") {
+        always_reset_initial_pose_ = parameter.as_bool();
+      } else if (param_name == "first_map_only") {
+        first_map_only_ = parameter.as_bool();
+      }
+    } else if (param_type == ParameterType::PARAMETER_INTEGER) {
+      if (param_name == "max_beams") {
+        max_beams_ = parameter.as_int();
+        reinit_laser = true;
+      } else if (param_name == "max_particles") {
+        max_particles_ = parameter.as_int();
+        reinit_pf = true;
+      } else if (param_name == "min_particles") {
+        min_particles_ = parameter.as_int();
+        reinit_pf = true;
+      } else if (param_name == "resample_interval") {
+        resample_interval_ = parameter.as_int();
+      }
+    }
+  }
+
+  // Checking if the minimum particles is greater than max_particles_
+  if (min_particles_ > max_particles_) {
+    RCLCPP_WARN(
+      this->get_logger(),
+      "You've set min_particles to be greater than max particles,"
+      "this isn't allowed so they'll be set to be equal.");
+    max_particles_ = min_particles_;
+  }
```

Comment:
I am not sure what should be done here in case of an invalid min/max particule set request:
- Set to equal and return success = true
- Set to equal and return success = false
- Not changing the parameter and return success = false ?

Reply:
Not change and return failure